### PR TITLE
txn: Use TransactionOption in store/tikv module

### PR DESCRIPTION
**Pull request by longfangsong**
### What problem does this PR solve?

This is a step forward of #23130.

We introduced TransactionOption there, which can also be used in store/tikv module.

### What is changed and how it works?

unify all the newTiKVTxnXXX into just `newTiKVTxnWithOptions`, and add a `DefaultTransactionOption` to make our life easier in tests.

### Check List 

Tests 

- Unit test

### Release note

- No release note

### CODE REVIEW

**Comment by longfangsong** [comment]
@Yisaer PTAL

File: store/tikv/txn.go [hunk 1]
```diff
@@ -69,41 +69,47 @@ type KVTxn struct {
...
 }
 
-// newTiKVTxnWithStartTS creates a txn with startTS.
-func newTiKVTxnWithStartTS(store *KVStore, txnScope string, startTS uint64, replicaReadSeed uint32) (*KVTxn, error) {
-	ver := kv.NewVersion(startTS)
-	snapshot := newTiKVSnapshot(store, ver, replicaReadSeed)
+func newTiKVTxnWithOptions(store *KVStore, options kv.TransactionOption) (*KVTxn, error) {
+	if options.TxnScope == "" {
+		panic("TxnScope unfilled!")
+	}
```
Comment (longfangsong) [inline store/tikv/txn.go:93]: Do we have anything like `assert!` in TiDB?

Comment (Yisaer) [reply store/tikv/txn.go:93]: If `options.TxnScope` is "", we use `global` instead.

**Review by Yisaer — COMMENTED** [review]
Rest LGTM, plz fix the conflict.

File: store/tikv/txn.go [hunk 2]
```diff
@@ -83,44 +87,83 @@ type KVTxn struct {
 	kvFilter           KVFilter
 }
 
-func newTiKVTxn(store *KVStore, txnScope string) (*KVTxn, error) {
-	bo := NewBackofferWithVars(context.Background(), tsoMaxBackoff, nil)
-	startTS, err := store.getTimestampWithRetry(bo, txnScope)
-	if err != nil {
-		return nil, errors.Trace(err)
+func extractStartTs(store *KVStore, options kv.TransactionOption) (uint64, error) {
```
Comment (Yisaer) [inline store/tikv/txn.go:90]: Add a unit test to for `extractStartTs`

**Review by Yisaer — COMMENTED** [review]
Rest LGTM

**Review by Yisaer — APPROVED** [review]
/lgtm

**Review by xhebox — APPROVED** [review]

**Pull request merged**